### ISSUE TITLE
Unable to Execute Read Queries During Table Loading - Request for Concurrent Loading and Querying

### ISSUE THREAD

**Issue by mrchypark** [roapi/roapi#339]
Thank you for creating the roapi project! I'm currently working on a project using OSS and Parquet, and I have a question.

We can't execute read queries while tables are being loaded. This seems like a significant limitation, so I wanted to raise it as an issue.

We're using the most recent release, **v0.11.3**.

Are there any updates related to this?

**Comment by houqp** [comment]
Table only gets loaded once on startup, or are you referring to loading tables dynamically at runtime blocking new queries?

**Comment by mrchypark** [comment]
Yes! to loading tables dynamically at runtime blocking new queries.

**Comment by houqp** [comment]
The basic infrastructure to allow general concurrent table reload has now landed: https://github.com/roapi/roapi/pull/360. The remaining task is to replace the current reloader code with the new infra:

https://github.com/roapi/roapi/blob/1570c7d4bb433aa71a957b03236f1c78e5a667b6/roapi/src/startup.rs#L31
